### PR TITLE
remove unnecessary defer function wrappers

### PR DESCRIPTION
**Pull request by lucming**
## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: libcalico-go/lib/ipam/ipam.go [hunk 1]
```diff
@@ -992,16 +992,13 @@ func (c ipamClient) ReleaseIPs(ctx context.Context, ips ...ReleaseOptions) ([]ne
 		switch ip.Version() {
 		case 4:
 			pool, err = c.blockReaderWriter.getPoolForIP(*ip, v4Pools)
-			if err != nil {
```
Comment (caseydavenport) [inline libcalico-go/lib/ipam/ipam.go:995]: This is the only diff here that I would prefer to revert - we like to keep error handling immediately after the call that generates the error. It's slightly more verbose but is less prone to unexpected bugs in the future as this code changes.

**Review by caseydavenport — COMMENTED** [review]
Looks good, but there's one diff that I'd like to revert.

File: libcalico-go/lib/ipam/ipam.go (hunk 1 above)
Comment (lucming) [reply libcalico-go/lib/ipam/ipam.go]: > This is the only diff here that I would prefer to revert - we like to keep error handling immediately after the call that generates the error. It's slightly more verbose but is less prone to unexpected bugs in the future as this code changes.

okay, I've recovered

**Review by caseydavenport — APPROVED** [review]

**Comment by caseydavenport** [comment]
Very sorry this didn't get merged. Looks like it slipped through the cracks and is now conflicting, so will need a rebase.

Given how old it is, I'm going to close it but please shout if you'd like it reopened.

**Pull request closed without merging**